### ISSUE TITLE
Feature request: subversion module should handle owner, group and permissions

### ISSUE THREAD

**Issue by Petrox** [ansible/ansible-modules-core#2515]
Like the file and copy module does.

ubuntu 14.04
ansible 1.9.4
subversion 1.8.8

**Comment by jimi-c** [comment]
Generally the scm modules all rely on the remote_user (or become_user) to control ownership. Specifying permissions might be useful, though most users can control that currently via the umask setting.

**Comment by Petrox** [comment]
I agree that this is a feature request, to make life easier to the admin (as that is the single most important job of ansible is).

Become could solve the ownership in simple situations, but there might be situations where the given user could not create a working copy in the given path (like /var/www ), so it's unsufficient.

Become does not solve the problem of groups and permissions. Of course we could do that in a separate task. But then, we could do that in a separate task when using the copy module, but it does handle these.

So of course there's a workaround, as one can create the directory as root, set the ownership, then become-subversion-checkout there, then set the groups and permissions recursively.

These could be in the same task: "owner=somedeveloper group=developers"

Subversion ansible module does not have an umask setting: http://docs.ansible.com/ansible/subversion_module.html

google search "ansible umask" does not even return *.ansible.com results on the first page, and noone will visit the second. :)

Cheers :)

**Comment by ansibot** [comment]
@dsummersl, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/jctanner/ansibullbot/blob/ISSUE_TRIAGE_CLASS/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@dsummersl, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@dsummersl, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@dsummersl, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by dsummersl** [comment]
I notice that the git module supports a `umask` feature in 2.2 so there is some precedence for this - although bzr/hg don't.

**Comment by ansibot** [comment]
This repository has been locked. All new issues and pull requests should be filed in https://github.com/ansible/ansible

Please read through the <a href="http://docs.ansible.com/ansible/dev_guide/repomerge.html#move-issues-and-prs-to-new-repo">repomerge</a> page in the dev guide. The guide contains links to tools which automatically move your issue or pull request to the ansible/ansible repo.

**Comment by ansibot** [comment]
This issue was migrated to https://github.com/ansible/ansible/issues/29992
